### PR TITLE
Fixing crash case

### PR DESCRIPTION
**Pull request by CoalSephos**
This should resolve an issue causing the bot to crash upon not resolving a user correctly from cached users.

### CODE REVIEW

File: src/data/threads.js [hunk 1]
```diff
@@ -117,7 +116,7 @@ async function createNewThreadForUser(user, quiet = false) {
     displayNote = `[${note.created_at}] (${note.created_by_name}) => **Note:** ${note.note}\n`;
   } else
     displayNote = '';
-  const infoHeader = `NAME **${mainGuildNickname}**\nMENTION ${member ? member.mention : 'UNKNOWN'}\nID **${user.id}**\nACCOUNT AGE **${accountAge}**\n`
+  const infoHeader = `NAME **${mainGuildNickname}**\nMENTION ${member ? member.mention || user.mention : 'UNKNOWN'}\nID **${user.id}**\nACCOUNT AGE **${accountAge}**\n`
```
Comment (bsian03) [inline src/data/threads.js:119]: `member.mention` will definitely exist if the ternary passes

File: src/data/threads.js [hunk 2]
```diff
@@ -97,17 +97,16 @@ async function createNewThreadForUser(user, quiet = false) {
...
+  if (! member) console.log(`[INFO] Member ${user.id} not found in main guild ${config.mainGuildId}`);
 
   let mainGuildNickname = null;
-
-  if(member && member.nick) mainGuildNickname = member.nick
-  else if(member && ! member.nick) mainGuildNickname = member.username
-  else if(! member) mainGuildNickname = 'An unknown user'
-
+  if (member && member.nick) mainGuildNickname = member.nick;
+  else if (member && member.user) mainGuildNickname = member.user.username;
```
Comment (bsian03) [inline src/data/threads.js:106]: `.user` will always exist

File: src/data/threads.js [hunk 3]
```diff
@@ -97,17 +97,16 @@ async function createNewThreadForUser(user, quiet = false) {
...
+  else if (member == null) mainGuildNickname = 'NOT ON SERVER';
+  if (mainGuildNickname == null) mainGuildNickname = 'UNKNOWN';
```
Comment (bsian03) [inline src/data/threads.js:108]: why not just else this?

File: src/data/threads.js [hunk 4]
```diff
@@ -97,17 +97,12 @@ async function createNewThreadForUser(user, quiet = false) {
...
-  if (! member)
-    return console.log(`[INFO] Member ${user.id} not found in main guild ${config.mainGuildId}`);
-
-  let mainGuildNickname = null;
-
-  if(member && member.nick) mainGuildNickname = member.nick
-  else if(member && ! member.nick) mainGuildNickname = member.username
-  else if(! member) mainGuildNickname = 'An unknown user'
+  const mainGuild = utils.getMainGuild();
+  const member = (mainGuild ? await bot.getRESTGuildMember(mainGuild.id, user.id) : null);
```
Comment (bsian03) [inline src/data/threads.js:101]: What's with the end brackets?

Comment (CoalSephos) [reply src/data/threads.js:101]: no idea i didn't put them there

File: src/data/threads.js [hunk 5]
```diff
@@ -97,17 +97,12 @@ async function createNewThreadForUser(user, quiet = false) {
...
-  if (! member)
-    return console.log(`[INFO] Member ${user.id} not found in main guild ${config.mainGuildId}`);
-
-  let mainGuildNickname = null;
-
-  if(member && member.nick) mainGuildNickname = member.nick
-  else if(member && ! member.nick) mainGuildNickname = member.username
-  else if(! member) mainGuildNickname = 'An unknown user'
+  const mainGuild = utils.getMainGuild();
+  const member = mainGuild ? await bot.getRESTGuildMember(mainGuild.id, user.id) : null;
```
Comment (bsian03) [inline src/data/threads.js:101]:
```suggestion
  const member = await this.bot.getRESTGuildMember(mainGuild.id, user.id).catch(e => null);
```

**Pull request merged**